docs/migrate-matomo-4-to-5.md

```diff
@@ -0,0 +1,128 @@
+---
+category: Develop
+---
+# Migrate Plugin from Matomo 4.X to Matomo 5
+
+This migration guide covers how to do some migrations to make a plugin compatible with Matomo 5. A list of
+all changes in Matomo 5 can be found in the [Changelog](/changelog).
+
+## Create a new branch
+
+We recommend creating a new branch for your plugin that supports Matomo 5. For example `5.x-dev`. This way you will be able to make changes to your plugin for Matomo 4 and Matomo 5 and release independent versions for each of them. You can still publish updates to your plugin that supports Matomo 4 once you have published an update for a version that supports Matomo 5.
+
+## Adjust the required Matomo version
+
+For your plugin to be executed in Matomo 5 you first need to show it is compatible with Matomo 5 in your `plugin.json` file:
+
+* specify that your plugin requires Matomo 5 (the requirement for Matomo 4 used to be e.g. `"matomo": ">=4.0.0-b1,<5.0.0-b1"`). 
+* we also recommend increasing your plugin's major version number e.g. from `4.1.9` to `5.0.0`.
+
+The `plugin.json` would look like this:
+
+```json
+   "version": "5.0.0",
+   "require": {
+        "matomo": ">=5.0.0-b1,<6.0.0-b1"
+    },
+```
+
+It's not allowed to support multiple major Matomo versions such as Matomo 4 and Matomo 5: `"matomo": ">=4.0.0-b1,<6.0.0-b1"`. You would in this case receive an error email and the release would not be published.
+
+## Required PHP version
+
+Matomo still requires PHP 7.2.5. So there won't be any changes needed in terms of e.g. removed PHP functions. But with releases of PHP 8.1 and 8.2 you may need to check if your plugin might not start throwing warnings or deprecation notices in newer PHP releases.
+
+## AngularJS replaced with VueJS
+
+Matomo 5 completes the migration from AngularJS to the [VueJS 3 framework](https://vuejs.org/guide/introduction.html). If your plugin contains templates which make use of AngularJS components or directives then these will need to be recreated in VueJS and the templates adjusted. 
+
+A new VueJS component can be created for your plugin using the console command:
+
+```
+./console generate:vue-component --pluginname=MYPLUGIN --component=NEWCOMPONENT
+``` 
+
+This will create an example component in `plugins/MYPLUGIN/vue/src/NEWCOMPONENT.vue` which shows the basic component structure and can be used a starting point to build a new component.
```

Comment:
```suggestion
This will create an example component in `plugins/MYPLUGIN/vue/src/NEWCOMPONENT.vue` which shows the basic component structure and can be used as a starting point to build a new component.
```